pkg/virt-handler/hotplug-disk/findmnt_test.go

```diff
@@ -119,19 +119,38 @@ var _ = Describe("findmnt", func() {
 		table.Entry("for findmntbydevice", callFindMntByDeviceInvalidJson),
 	)
 
-	It("GetSource should properly match source field", func() {
+	It("GetSourcePath should properly match source field", func() {
 		test := FindmntInfo{
 			Source: "/dev/test",
 		}
-		Expect(test.GetSource()).To(Equal("/dev/test"))
+		Expect(test.GetSourcePath()).To(Equal("/dev/test"))
 		test2 := FindmntInfo{
 			Source: "/dev/test[/mnt/something/else/]",
 		}
-		Expect(test2.GetSource()).To(Equal("/mnt/something/else/"))
+		Expect(test2.GetSourcePath()).To(Equal("/mnt/something/else/"))
 		test3 := FindmntInfo{
 			Source: "/dev/test[/mnt/something/else/[/more]]",
 		}
-		Expect(test3.GetSource()).To(Equal("/mnt/something/else/[/more]"))
+		Expect(test3.GetSourcePath()).To(Equal("/mnt/something/else/[/more]"))
+	})
+
+	It("GetSourceDevice should return the device", func() {
+		test := FindmntInfo{
+			Source: "/dev/test",
+		}
+		Expect(test.GetSourceDevice()).To(Equal(""))
```

Comment:
why wouldn't this test case return `/dev/test` ?

Reply:
Because the regex is looking for .*[.+] so no [ ] is no match. This is because I am calling findmnt with arguments that should never return /dev/test for a filesystem mount.